packages/webchat/src/main.tsx

```diff
@@ -168,12 +190,15 @@ class Web extends React.Component<MainProps> {
     })
 
     observe(this.props.config!, 'overrides', (data: any) => {
+      // TODO: 12 - we didn't check if window.parent was set the first time
+      // we loaded overrideds, so why do we do now?
       if (data.newValue && window.parent) {
         this.loadOverrides(data.newValue)
       }
     })
 
     observe(this.props.dimensions!, 'container', (data: any) => {
+      // TODO: 13 - the first time we do setWith we do it with props.containerWidth. Why the sudden change to props.dimensions?
       if (data.newValue && window.parent) {
         this.postMessageToParent('setWidth', data.newValue)
```

Comment:
containerWidth / layoutWidth are the configurations you can change when initializing the webchat. Here we listen if the dimension of the container changes, if it does, we call the parent iframe to resize correctly so the webchat is displayed correctly.

Behind the scene, we store both values in props.dimensions (container/layout).